### DISCUSSION TITLE
Thermal hydraulics module problem

### DISCUSSION THREAD

**Discussion by YuhangNIU** [idaholab/moose#21641] [Q&A Modules: General] [answered]
Hi all,

I want to know how the pipe component works in the thermal-hydraulic module. So I try to restore an input file (clg.densityvelocity_3eqn.i 
 “[clg.densityvelocity_3eqn.txt](https://github.com/idaholab/moose/files/9147933/clg.densityvelocity_3eqn.txt)”) to a normal form “[pipe.txt](https://github.com/idaholab/moose/files/9147934/pipe.txt)” (without encapsulation). But something goes wrong. Could anyone kindly help me to understand it?

![Screenshot from 2022-07-20 03-17-45](https://user-images.githubusercontent.com/80933114/179933971-ec910681-064a-4651-a277-8d202409eb04.png)

Best regards,
Yuhang Niu

**Comment by GiudGiud** [comment, marked as answer]
@licharlot seg-fault in thm, might be worth a look

**Reply by licharlot** [reply]
Hi Yuhang, 
Could you please give us more details about what you are trying to understand/do? THM is designed to be used with the `Components` syntax. Thanks.

**Reply by joshuahansel** [reply]
Yuhang, just to clarify, we're trying to see if there's an easier solution to what you're trying to do, and we still are interested in knowing. However, I did look into your input file and found the segfault is occurring in `ADRDG3EqnMaterial`. You directly provide values for the `A_linear` and `A_elem` parameters, but the parameters are actually variable names (MOOSE allows values to be used in place of variable names). The segfault occurs because we make a call `getVar("A_elem", 0)`, for example. To fix your input file, add
```
[AuxVariables]
  [A_elem]
    initial_condition = 1.907720E-04
  []
  [A]
    initial_condition = 1.907720E-04
  []
[]
```
and then in your `GlobalParams`:
```
  A_elem = A_elem
  A_linear = A
```

**Reply by YuhangNIU** [reply]
Thank you for your reply. Now, it works.
